full-stack-tests/presentation/src/frontend/content/ContentDisplayLabels.test.ts

```diff
@@ -0,0 +1,195 @@
+      <ECSchemaReference name="BisCore" version="1.0.0" alias="bis"/>
+      <ECEntityClass typeName="Color">
+        <BaseClass>bis:ElementMultiAspect</BaseClass>
+        <ECProperty propertyName="R" typeName="int" />
+        <ECProperty propertyName="G" typeName="int" />
+        <ECProperty propertyName="B" typeName="int" />
+      </ECEntityClass>
+    `;
+    imodel = await buildTestIModelConnection(this.test!.title, async (db) => {
+      const schema = importSchema(this, db, schemaXml);
+      schemaName = schema.schemaName;
+      const { id: modelId } = insertPhysicalModelWithPartition({ db, codeValue: "Model" });
+      const { id: categoryId } = insertSpatialCategory({ db, codeValue: "Category" });
+      for (let i = 0; i < 5; ++i) {
+        const { id: elementId } = insertPhysicalElement({
+          db,
+          modelId,
+          categoryId,
+          classFullName: elementClassName,
+        });
+
+        insertElementAspect({
+          db,
+          classFullName: schema.items.Color.fullName,
+          elementId,
+          ["R"]: i,
+          ["G"]: i + 1,
+          ["B"]: i + 2,
+        });
+
+        insertElementAspect({
+          db,
+          classFullName: schema.items.Color.fullName,
+          elementId,
+          ["R"]: i + 100,
+          ["G"]: i + 101,
+          ["B"]: i + 102,
+        });
+      }
+    });
+  });
+
+  after(async () => imodel?.close());
+
+  function createRuleset(...additionalRules: Rule[]): Ruleset {
+    return {
+      id: Guid.createValue(),
+      rules: [
+        {
+          ruleType: RuleTypes.Content,
+          specifications: [
+            {
+              specType: ContentSpecificationTypes.ContentInstancesOfSpecificClasses,
+              classes: { schemaName: "Generic", classNames: ["PhysicalObject"] },
+              relatedProperties: [
+                {
+                  propertiesSource: {
+                    direction: "Forward",
+                    relationship: { schemaName: "BisCore", className: "ElementOwnsMultiAspects" },
+                    targetClass: { schemaName, className: "Color" },
+                  },
+                  properties: "*",
+                  relationshipMeaning: "RelatedInstance",
+                },
+              ],
+            },
+          ],
+        },
+        ...additionalRules,
+      ],
+    };
+  }
+
+  it("each related content item has display label", async () => {
+    const ruleset = createRuleset();
+    const { items } = (await Presentation.presentation.getContentIterator({
+      imodel,
+      rulesetOrId: ruleset,
+      descriptor: { displayType: DefaultContentDisplayTypes.Grid },
+      keys: new KeySet(),
+    }))!;
+
+    for await (const item of items) {
+      for (const value of Object.values(item.values)) {
+        if (!Value.isNestedContent(value)) {
+          continue;
+        }
+        expect(value.length).to.eq(2);
+        expect(value[0].labelDefinition).to.deep.eq(EMPTY_LABEL);
```

Comment:
why not check labels of both related content values?

Reply:
Fixed